### PR TITLE
Remove use of `URL` from `url`

### PR DESCRIPTION
**Pull request by zfben**
URL is a global reference, it doesn't need to import before use. And import will break webpack.

Here is a related issue:

https://github.com/mdx-js/mdx/issues/1802

### CODE REVIEW

File: packages/mdx/lib/plugin/recma-document.js [hunk 1]
```diff
@@ -39,7 +39,6 @@
  *   JSX runtime to use.
  */
 
-import {URL} from 'url'
```
Comment (wooorm) [inline packages/mdx/lib/plugin/recma-document.js:42]: There are likely other `URL` imports in other files.
So only changing this line, probably hides one error, but probably shows more errors.
Could you also change the other ones?

**Review by wooorm — COMMENTED** [review]
Thanks for your PR. Can you change the other `URL` imports too?

Did you also read the conversation you linked to? Because you can configure webpack.

Luckily, TypeScript was fixed to finally support a global `URL`: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/58277#issuecomment-1026571932. So this can now be changed.

File: website/mdx-config.js [hunk 2]
```diff
@@ -8,7 +8,6 @@
 
 import path from 'path'
 import process from 'process'
-import {pathToFileURL} from 'url'
```
Comment (wooorm) [inline website/mdx-config.js:11]: This is broken.

File: website/post.js [hunk 3]
```diff
@@ -1,5 +1,4 @@
 import {promises as fs} from 'fs'
-import {fileURLToPath} from 'url'
```
Comment (wooorm) [inline website/post.js:2]: This is broken.

File: website/prep.js [hunk 4]
```diff
@@ -1,6 +1,5 @@
 #!/usr/bin/env node
 import {promises as fs} from 'fs'
-import {fileURLToPath} from 'url'
```
Comment (wooorm) [inline website/prep.js:3]: This is broken.

File: website/prerender.js [hunk 5]
```diff
@@ -1,7 +1,6 @@
 #!/usr/bin/env node
 import {promises as fs} from 'fs'
 import path from 'path'
-import {fileURLToPath} from 'url'
```
Comment (wooorm) [inline website/prerender.js:4]: This is broken.

**Comment by zfben** [comment]
@wooorm I have removed all `URL`.
I know it can be through with configure webpack, but removing URL is a better way currently.
In fact, the first time for this issue is when I use mdx in a vite project.

**Review by wooorm — COMMENTED** [review]
Nice! Could you remove the changes to `package-lock.json` from this PR? Rest looks good!

**Comment by zfben** [comment]
> Nice! Could you remove the changes to `package-lock.json` from this PR? Rest looks good!

done

**Review by wooorm — APPROVED** [review]

**Comment by wooorm** [comment]
Thanks! :)

**Comment by wooorm** [comment]
Released in 2.1.0! :)

**Pull request merged**